### PR TITLE
[New Rule] Executable Masquerading as Kernel Process

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
Monitors for kernel processes with associated process executable fields that are not empty. Unix kernel processes such  as kthreadd and kworker typically do not have process.executable fields associated to them. Attackers may attempt to hide their malicious programs by masquerading as legitimate kernel processes.

### CODE REVIEW

**Review by imays11 — APPROVED** [review]

File: rules/linux/defense_evasion_kthreadd_masquerading.toml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+[rule]
+author = ["Elastic"]
+description = """
+Monitors for kernel processes with associated process executable fields that are not empty. Unix kernel processes such 
+as kthreadd and kworker typically do not have process.executable fields associated to them. Attackers may attempt to
+hide their malicious programs by masquerading as legitimate kernel processes.  
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
```
Comment (Samirbous) [inline rules/linux/defense_evasion_kthreadd_masquerading.toml:17]: is this rule compatible also with other indexes ?

**Review by Samirbous — APPROVED** [review]

File: rules/linux/defense_evasion_kthreadd_masquerading.toml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        "Domain: Endpoint",
+        "OS: Linux",
+        "Use Case: Threat Detection",
+        "Tactic: Defense Evasion",
+        "Data Source: Elastic Defend"
+        ]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and
```
Comment (Aegrah) [inline rules/linux/defense_evasion_kthreadd_masquerading.toml:63]:
```suggestion
process where host.os.type == "linux" and event.action in ("exec", "exec_event") and event.type == "start" and
```

File: rules/linux/defense_evasion_kthreadd_masquerading.toml (hunk 1 above)
Comment (Aegrah) [reply rules/linux/defense_evasion_kthreadd_masquerading.toml]: ++, added endgame & auditd_manager support. Will check others.

Comment (Aegrah) [reply rules/linux/defense_evasion_kthreadd_masquerading.toml]: When adding auditd_manager support, I noticed I **must** supply the `event.dataset` value in the query - this currently does not make sense. Will check with Terrance to see whether we can extend coverage without being forced to do this. Created this Meta for it: https://github.com/elastic/detection-rules/issues/3428.

**Pull request merged**